library/src/main/java/com/q42/q42stats/library/Q42Stats.kt

```diff
@@ -1,5 +1,74 @@
 package com.q42.q42stats.library
 
-class Q42Stats {
-    val name = "Q42 Stats lib"
+import android.content.Context
+import com.q42.q42stats.library.collector.AccessibilityCollector
+import com.q42.q42stats.library.collector.PreferencesCollector
+import com.q42.q42stats.library.collector.SystemCollector
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.io.Serializable
+import java.util.concurrent.atomic.AtomicBoolean
+
+const val TAG = "Q42Stats"
+
+private val MUTEX = Unit
+
+class Q42Stats(private val config: Q42StatsConfig) {
+    private val isRunning = AtomicBoolean(false)
+
+    /* Collects stats and sends it to the server. This method is safe to be called from anywhere
+    in your app and will do nothing if it has already run before */
+    fun runAsync(context: Context) {
+        Q42StatsLogger.d(TAG, "Q42Stats: Checking Preconditions")
+        if (isRunning.get()) {
```

Comment:
this is confusing to read / strange usage. Normally you would use getAndSet or similar functions if you use an AtomicBoolean, and then you don't need the synchronized block. But see comment further on: using coroutines might be best.